fastparquet/writer.py

```diff
@@ -1486,3 +1487,52 @@ def sort_key(row_group) -> int:
 def write_thrift(f, obj):
     # TODO inline this
     return f.write(obj.to_bytes())
+
+def update_file_metadata(path: str, custom_metadata: dict,
+                         is_metadata: bool = None):
+    """Update metadata in file without rewriting data portion if a data file.
+
+    Update strategy depends if key found in new custom metadata is also found
+    in already existing custom metadata within thrift object, as well as its
+    value.
+        
+      - If not found in existing, it is added.
+      - If found in existing, it is updated.
+      - If its value is `None`, it is not added, and if found in existing,
+        it is removed from existing.
+
+    Parameters
+    ----------
+    path : str
+        Path to file.
+    custom_metadata : dict
+        Key-value metadata to update in thrift object.
+    is_metadata : bool, default None
+        Define if target file is a pure metadata file, or is a parquet data
+        file. Depending file type, metadata is not stores at same location.
```

Comment:
> metadata is not stores at same location.

not quite true: it's always at the *end*, it just that for the metadata files there is no data, so we know that the end starts just four bytes in.

Reply:
Ok, I removed this part of the comment:
_Depending file type, metadata is not stores at same location._

And I added as comment directly where processing to pure _metadata_ file is done the comment:
_# For pure metadata file, metadata starts just four bytes in._